### DISCUSSION TITLE
Records & Tuples

### DISCUSSION THREAD

**Discussion by corasaurus-hex** [clavascript/clavascript#38] [Ideas]
One complaint I heard in the cherry channel was how clojurescript had fallen behind with interop. Given that, it might make sense to plan for likely js features coming down the road so that interop doesn't fall behind.

One of these features that I think will be important is the proposal for Records & Tuples: https://tc39.es/proposal-record-tuple/tutorial/

I'm not sure we need to _do_ anything right now but keeping it in mind for designing interop in general seems like a smart move to me.

We can probably close this issue right away but if we want to discuss then this is a good place to do it.

**Comment by borkdude** [comment]
@corasaurus-hex Converted this into a Discussion. Let's reserve issues for things that are more or less ready to work on (or contain summaries of longer discussions) and let's use Discussions where we can go into depths.

**Comment by borkdude** [comment]
We can support the creation of records and tuples via a reader macro, e.g. `#rec {:a 1}` and `#tup [1 2 3]`. Other than that, I think `assoc` etc should perhaps already work and if not, we can make them work with these news types (if they will ever see the light of day).

**Comment by PEZ** [comment]
Would it be possible to use something like babel to add Records and Tuples to ClavaScript from the get-go? If so reader macros could be used for creating mutable objects. Then literal arrays and things could default to the immutable versions. (I really don't know what I am talking about, I'm just curious and a bit in love with the ”immutable first” legacy of Clojure's.)

**Reply by borkdude** [reply]
I would say this is too experimental: the proposal may still be rejected and also these collections aren't the default in JS today. The philosophy of clava is to embrace the JS runtime as it is now, without adding extra bundle size and/or tooling.
